File: codeforces/practice/264A/stones.cpp
#include <bits/stdc++.h>

using namespace std;

int arr[1000001];
int main() {
  string s;
  cin >> s;
  int i = 0;
  int right = s.size() - 1;
  int left = 0; 
  while(s[i]!='\0') {
    switch(s[i]) {
      case 'l':
        arr[right] = i + 1;
        right--;
        break;
      case 'r':
        arr[left] = i + 1;
        left++;    
        break;
    }
    i++;
  }

  for(int i=0; i<s.size(); i++) {
    printf("%d\n", arr[i]);
  }
  return 0;
}

File: codeforces/practice/264A/stones.scala
import scala.io.StdIn._

object Main {
  def main(args: Array[String]): Unit = {
    val s = readLine
    val res = new Array[Int](s.size)
    var right = s.size - 1
    var left = 0
    var i = 0
    while(i < s.size) {
      if (s(i) == 'l') {
        res(right) = i + 1
        right -= 1
      } else {
        println(i + 1)  
        left += 1
      }
      i += 1
    }
    var k = left
    while (k<s.size) {
      println(res(k))
      k += 1
    }
  }
}
